src/peergos/shared/user/ScryptGenerator.java

```diff
@@ -11,14 +11,22 @@
 
     @JsProperty
     public final int memoryCost, cpuCost, parallelism, outputBytes;
+    @JsProperty
+    public final String extraSalt;
 
-    public ScryptGenerator(int memoryCost, int cpuCost, int parallelism, int outputBytes) {
+    public ScryptGenerator(int memoryCost, int cpuCost, int parallelism, int outputBytes, String extraSalt) {
         if (memoryCost < MIN_MEMORY_COST)
```

Comment:
Can we either add a comment or more information to the exception message about MIN_MEMORY_COST? Is it 17 bytes? Why that number?

Reply:
17 is the memory factor in the [scrypt](https://en.wikipedia.org/wiki/Scrypt) hashing function. It means the calculation will use 2^17 bytes of memory. This number was originally chosen to be sufficiently strong against brute force attacks, and to still take ~1s on a mobile.

Reply:
👍